packages/server-side-render/src/server-side-render.tsx

```diff
@@ -16,27 +21,45 @@ import { useSelect } from '@wordpress/data';
  * Internal dependencies
  */
 import { useServerSideRender } from './hook';
+import type { UseServerSideRenderArgs } from './hook';
 
 const EMPTY_OBJECT = {};
 
-function DefaultEmptyResponsePlaceholder( { className } ) {
+interface PlaceholderProps {
+	className?: string;
+}
+
+interface ErrorPlaceholderProps extends PlaceholderProps {
+	message?: string;
+}
+
+interface LoadingPlaceholderProps {
+	children?: ReactNode;
+}
+
+function DefaultEmptyResponsePlaceholder( { className }: PlaceholderProps ) {
 	return (
 		<Placeholder className={ className }>
 			{ __( 'Block rendered as empty.' ) }
 		</Placeholder>
 	);
 }
 
-function DefaultErrorResponsePlaceholder( { message, className } ) {
+function DefaultErrorResponsePlaceholder( {
+	message,
+	className,
+}: ErrorPlaceholderProps ) {
 	const errorMessage = sprintf(
 		// translators: %s: error message describing the problem
 		__( 'Error loading block: %s' ),
-		message
+		message || 'Unknown error'
```

Comment:
This should be translated

```suggestion
		message || __( 'Unknown error' )
```

Reply:
Another approach would be to get rid of this extra string by not making `message` optional in `ErrorPlaceholderProps` and then using `message={ error! }` in `<ErrorResponsePlaceholder />`. We can assert that because the description for `error` mentions `"The error message (available when status is 'error')."`, which is already the case when it's used there.